src/pages/ShopPage/components/HistoryContent/index.js

```diff
@@ -1,98 +1,140 @@
+  selectHistoryStatus,
+} from 'features/history/historySlice'
+import { dataLoadingStatus } from 'hooks/dataLoadingStatus'
 
 const HistoryContent = () => {
-  const useHistory = API.useHistory()
-
-  const useGroceriesCategories = API.useGroceriesCategories()
-  const useProductsCategories = API.useProductsCategories()
-  const dishesCategories = API.UseDishesCategory()
-  const useSavedListsCategories = API.useSavedListsCategories()
-
-  const [mutate_DELETE_DeleteHistory] = API.useDeleteHistory()
-
-  if (
-    useHistory.isError ||
-    useGroceriesCategories.isError ||
-    useProductsCategories.isError ||
-    dishesCategories.isError ||
-    useSavedListsCategories.isError
-  ) {
-    return 'Fetching date error...'
-  } else if (
-    useHistory.isLoading ||
-    useGroceriesCategories.isLoading ||
-    useProductsCategories.isLoading ||
-    dishesCategories.isLoading ||
-    useSavedListsCategories.isLoading
-  ) {
-    return 'Loading date...'
-  } else if (
-    useHistory.isSuccess &&
-    useGroceriesCategories.isSuccess &&
-    useProductsCategories.isSuccess &&
-    dishesCategories.isSuccess &&
-    useSavedListsCategories.isSuccess
-  ) {
-    const showWhenEmpty = (
-      <Styles.Message>Your purchase history is empty!</Styles.Message>
-    )
-
-    if (useHistory.data.length < 1) {
-      return showWhenEmpty
-    }
-
-    const CreateHistoryElement = useHistory.data.map(
-      ({ groceries, products, dishes, savedLists, id, DateToShow }) => {
-        return (
-          <Styles.Wrapper key={id}>
-            <div>
-              <Styles.Date>{DateToShow}</Styles.Date>
-              <Styles.Button
-                onClick={() =>
-                  mutate_DELETE_DeleteHistory({
-                    id: id,
-                  })
-                }
-              >
-                Delete from history
-              </Styles.Button>
-            </div>
-
-            <HistoryParentCategoryList
-              parentsTitle="Groceries"
-              itemsList={groceries}
-              filtredCategories={useGroceriesCategories.data}
-            />
-            <Styles.EmptyDivToSpace />
-            <HistoryParentCategoryList
-              parentsTitle="Products"
-              itemsList={products}
-              filtredCategories={useProductsCategories.data}
-            />
-            <Styles.EmptyDivToSpace />
-            <HistoryParentCategoryList
-              parentsTitle="Dishes"
-              itemsList={dishes}
-              filtredCategories={dishesCategories.data}
-            />
-            <Styles.EmptyDivToSpace />
-            <HistoryParentCategoryList
-              parentsTitle="SavedList"
-              itemsList={savedLists}
-              filtredCategories={useSavedListsCategories.data}
-            />
-          </Styles.Wrapper>
-        )
-      },
-    )
-
-    return CreateHistoryElement
+  const dispatch = useDispatch()
+  const [updateHistory, setUpdateHistory] = useState(false)
```

Comment:
jw. - do wyrzucenia

Reply:
Aaa już wiem czemu tak robiłem bo gdy było np tak : onClick={() => {
                dispatch(deleteHistoryBasket({ id: id }))
                dispatch(fetchHistoryBasket())
              }} 
to zdarzało się że dane zostawały pobrane zanim zostały zmienione

Reply:
w thunku:
- najpierw wołasz akcję, że zaczyna coś robić z danymi i ustawia status na loading,
- usuwa (await)
- pobiera nowe (await) (ale najlepiej zeby sama funkcja usuwania zwróciła nowy zestaw - tak się powinno robić przy DELETE)
- ustawienie state na podstawie pobranych danych i status na success

w komponencie tylko dajesz sygnał po kliknięciu że ma usunąć. Sam thunk ogarnia resztę

Reply:
przerobiłem również put i post metody tak aby wewnątrz nich następował fetch elementów.
Done.

Reply:
Dodałem jeszcze fajne helpery 👍 
Done.